app/controllers/works_controller.rb

```diff
@@ -0,0 +1,95 @@
+class WorksController < ApplicationController
+
+    def index
+        @works = Work.all
+    end
+
+    def show
+        @work = Work.find_by(id: params[:id])
+
+        if @work.nil?
+            head :not_found
```

Comment:
Consider adding redirect behavior rather than `head :not_found` to redirect the user to a useful page.